src/components/BurgerMenu/BurgerMenu.tsx

```diff
@@ -0,0 +1,87 @@
+import { useEffect } from 'react';
+
+import clsx from 'clsx';
+import Image from 'next/image';
+
+import { ButtonLink } from '~components/Buttons/ButtonLink';
+import { Icon } from '~components/Icon/Icon';
+import { CustomLink } from '~components/Link/Link';
+import { useScreenQuery } from '~hooks/useScreenQuery';
+
+import { navigation } from '../RootLayout/navigation';
+import logo from '/public/svg/logo.svg';
+
+import s from './BurgerMenu.module.scss';
+
+interface BurgerMenuProps {
+	onMenuOpen: (newState: boolean) => void;
+	isMenuOpen: boolean;
+}
+
+export function BurgerMenu({ onMenuOpen, isMenuOpen }: BurgerMenuProps) {
+	useEffect(() => {
+		const handleKeyDown = (e: KeyboardEvent) => {
+			if (e.code === 'Escape') {
+				onMenuOpen(false);
+			}
+		};
+		window.addEventListener('keydown', handleKeyDown);
+		return () => {
+			window.removeEventListener('keydown', handleKeyDown);
+		};
+	}, [onMenuOpen]);
+
```

Comment:
When burgerMenu is opend, need to fix scroll the main page.
For this good to use hook (https://github.com/AmelianceSkyMusic/ameliance-ui/blob/main/hooks/useScrollLock.ts)

Reply:
For discussion

Reply:
You are right, I will fix it